container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/engine_functions/create_engine.go

```diff
@@ -100,6 +108,18 @@ func CreateEngine(
 		return nil, stacktrace.Propagate(err,
 			"An error occurred attempting to create logging components for engine with GUID '%v' in Docker network with network id '%v'.", engineGuidStr, targetNetworkId)
 	}
+
+	// schedule log removal for log retention
+	go func() {
+		osFs := volume_filesystem.NewOsVolumeFilesystem()
+		realTime := logs_clock.NewRealClock()
+		logRemover := log_remover.NewLogRemover(osFs, realTime)
+
+		for range logRemovalTicker.C {
```

Comment:
Reading the docs of time.Ticket, it seems the ticker will first wait for the duration to elapse before ticking for the first time. So if users always uses the engine for less than 6 hours, you'll never cleanup the logs. We probably want to do a first cleanup immediately when the engine starts